workspaces/signals.py

```diff
@@ -1,33 +1,47 @@
-import traceback
-from django.db.models.signals import post_save
+from django.core.exceptions import ValidationError
+from django.db.models.signals import post_save, pre_save
 from django.dispatch import receiver
 from loguru import logger
 from safedelete.signals import post_softdelete
-import sentry_sdk
 
 from app_users.models import AppUser
+from workspaces.tasks import send_added_to_workspace_email
 from .models import Workspace, WorkspaceInvite, WorkspaceMembership, WorkspaceRole
 
 
-@receiver(post_save, sender=AppUser)
-def add_user_existing_workspace(instance: AppUser, **kwargs):
-    """
-    if the domain name matches, add the user to the workspace
-    """
-    if not instance.email:
+@receiver(pre_save, sender=AppUser)
+def invite_user_to_workspaces_with_matching_domain(instance: AppUser, **kwargs):
+    if instance.id or not instance.email:
+        # don't send invitation if: 1. user is being updated, 2. user doesn't have an email
```

Comment:
this does not allow existing users to be added automatically?

Reply:
that we do in `post_save`

Reply:
the invite will not exist?

Reply:
in any case - that code would not be in an `AppUser` signal, but after a workspace was updated

Reply:
e.g. I think this wont work.
1. create a workspace
2. signup as dev@dara.network
3. update workspace domain to dara.network
4. login as dev@dara.network 
5. ????

Reply:
that wouldn't have worked before this PR too

Reply:
Everytime you login, it saves the user so the old check would run no?

Reply:
also fixes the case when:
1. user is added to a workspace because domain name
2. user leaves workspace
3. user updates profile -> user is added to the workspace again

Reply:
and of course, doesn't use `create_and_send_invite` incorrectly

Reply:
> Everytime you login, it saves the user so the old check would run no?

why would login save the user?

Reply:
Ok - I thought we tracked user's last login but we apparently don't